compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2922,10 +2922,24 @@ impl<'a> Parser<'a> {
     }
 
     pub(crate) fn eat_label(&mut self) -> Option<Label> {
-        self.token.lifetime().map(|ident| {
+        if let Some(ident) = self.token.lifetime() {
+            // Disallow `'fn`, but with a better error message than `expect_lifetime`.
+            if ident.without_first_quote().is_reserved()
+                && ![kw::UnderscoreLifetime, kw::StaticLifetime].contains(&ident.name)
+            {
+                self.dcx().emit_err(errors::InvalidLabel { span: ident.span, name: ident.name });
+            }
+
+            // Disallow `'_: loop {}`
+            if [kw::StaticLifetime, kw::UnderscoreLifetime].contains(&ident.name) {
+                self.dcx().emit_err(errors::InvalidLabel { span: ident.span, name: ident.name });
+            }
```

Comment:
```suggestion
            if ident.without_first_quote().is_reserved() {
                self.dcx().emit_err(errors::InvalidLabel { span: ident.span, name: ident.name });
            }
```

Reply:
LOL I think those used to be separate error messages. Yeah, now it's the same, thanks for catching this oversight of mine.